### PR TITLE
feat: Style.Transform for altering strings at render time

### LINKED ISSUE #229: String transform method
It would be handy to have a method on a `Style` that can apply a function to a string:

```go
s := NewStyle().Transform(strings.ToUpper);
fmt.Println(s.Render("raow")) // RAOW
```

With the the argument there being a `func(string) string`.

### PR DESCRIPTION
**Pull request by meowgorithm**
So with this you can basically:

```go
s := NewStyle().Transform(strings.ToUpper)
fmt.Println(s.Render("raow!") // "RAOW!"
```

Some things to consider:

* Should it accept multiple functions?
* Is `Transform` the best name for this?
* Should it ignore ANSI sequences?

Closes #229.

### CODE REVIEW

**Review by maaslalani — APPROVED** [review]
Looks great to me, this will make things much more flexible!

**Comment by maaslalani** [comment]
1. I personally like the `Transform` naming.
2. I think we don't _have_ to take multiple transform functions since they are stackable i.e.:

```
s := NewStyle().Transform(func(s string) string {
  return strings.ToUpper(removeUnderline(prefix(strings.TrimSpace(s))))
})
```

but it could be nicer to have the following API:

```
s := NewStyle().Transform(strings.TrimSpace, prefix, removeUnderline, strings.ToUpper)
```

File: get.go [hunk 1]
```diff
@@ -408,6 +408,12 @@ func (s Style) GetFrameSize() (x, y int) {
 	return s.GetHorizontalFrameSize(), s.GetVerticalFrameSize()
 }
 
+// GetTransform returns the transform set on the style. If no transform is set
+// nil is returned.
+func (s Style) GetTransform() func(string) string {
```
[the remark this answers is not in the document]
Comment (meowgorithm) [reply get.go:413]: 100% — thanks for catching this!

**Comment by meowgorithm** [comment]
> I think we don't _have_ to take multiple transform functions since they are stackable i.e.:
> 
> ```go
> s := NewStyle().Transform(func(s string) string {
>   return strings.ToUpper(removeUnderline(prefix(strings.TrimSpace(s))))
> })
> ```
> 
> but it could be nicer to have the following API:
> 
> ```go
> s := NewStyle().Transform(strings.TrimSpace, prefix, removeUnderline, strings.ToUpper)
> ```
> 
> We can always merge this in to get and then do the multiple function later as that would not be / cause a breaking change to the API.

Agreed; we can feel it out and add it if we feel we need it. Thanks, @maaslalani.

File: get.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (maaslalani) [reply get.go:413]: Added https://github.com/charmbracelet/lipgloss/pull/232/commits/9d6896cf3708d7283e08003ce2de493383602e34

**Comment by bashbunni** [comment]
Happy to play around with this before merging to provide some feedback in how it feels to work with in Glamour. Funny, Muesli and I were just talking about these features being missing from Lip Gloss for Glamour. I'm not quite there yet, so if you're in a rush to merge, then I can provide feedback at a later time. Otherwise happy to try it out over the next few weeks.

**Comment by meowgorithm** [comment]
That would be awesome, Bash. No hurry to merge over here. This actually came out of some discussions with Muesli during `table` development so it totally makes sense that this plays into new Glamour (should we call it Glamour Total Makeover or something?).

**Comment by bashbunni** [comment]
✨ Glossy Glamour ✨ 
ok perfect, will do

**Comment by maaslalani** [comment]
Should we merge this? It will be quite useful for list enumerations where we pass an enumerator (i.e `IV`) and pass a `EnumeratorStyle` (`NewStyle().Transform(func(s string) string { return "(" + s ")" }`) to allow many different configurations for the styling of the enumerator:

For example:

```
IV
IV.
(II)

a
a.
(a)
```

**Pull request merged**